### DISCUSSION TITLE
patches to create only 4 FBs on Raven

### DISCUSSION THREAD

**Discussion by jalavoui** [ChefKissInc/NootedRed#321] [General]
### macOS Version

Ventura

### What is your CPU's model?

raven

### Please describe the behaviour in detail.

maybe a coincidence but with this patch and by disabling firefox gfx software render and atm i got no firefox crashes

### What should've happened instead?

need to confirm this

**Comment by jalavoui** [comment]
the fix might have removed some os x check made by firefox opengl engine since atm no pages crashed.
hope is not a coincidence
this is not a bug - it's a very good thing

**Comment by jalavoui** [comment]
btw this was tested with and without bfix... (censored kext name)

**Comment by VisualEhrmanntraut** [comment]
Interesting. Btw, @jalavoui I have other bugs to fix in my TODO list as well:
<img width="225" alt="Screenshot 2024-11-05 at 15 06 44" src="https://github.com/user-attachments/assets/91e31777-c251-4ba7-b084-25ba422378e7">
And also a list of debug print methods that are empty of which to add back their prints:
<img width="151" alt="image" src="https://github.com/user-attachments/assets/a6c01d2b-96db-4bd9-95cb-d6f9542e88da">

**Comment by jalavoui** [comment]
this is actually a good spot to discuss nred with you. i'm gonna investigate if this patch really make a difference by using a unpatched version.

did you notice that from some time ago this pci message address error and the cursor error ?
the pci error is on 1st nred load
![Captura de ecrã 2024-11-05, às 20 04 17](https://github.com/user-attachments/assets/10ed4278-1c21-4436-a68c-f3c9cd16d6f8)

but as driver finish loading and redos setup this cursor error msg pops up

![Captura de ecrã 2024-11-05, às 20 04 34](https://github.com/user-attachments/assets/72a8a0fc-b40e-40dd-8a6e-5709cc5b49f7)

**Reply by VisualEhrmanntraut** [reply]
The "AMD Error" message is not related to the PCI messages, it means something tried to allocate a memory handle but passed a null pointer. I don't know where it comes from. The second one is just because it got called before the controller got enabled, it's most likely fine.

**Reply by VisualEhrmanntraut** [reply]
(If you want to talk more efficiently, create a Telegram and message me)

**Comment by jalavoui** [comment]
agree. i just tested the dam firefox without your patches and it didnt crash as usual. guess it fixed itself.
still the patch is very good and you should keep it

**Comment by jalavoui** [comment]
updating as the firefox opengl crashes come back...

dam thing
